assets/src/components/activities/directed-discussion/discussion/participation-util.tsx

```diff
@@ -0,0 +1,46 @@
+import { DDParticipationDefinition } from '../schema';
+import { ThreadedPost } from './discussion-service';
+import { flatPosts } from './post-util';
+
+export interface CurrentParticipation {
+  posts: number;
+  replies: number;
+  canPost: boolean;
+  canReply: boolean;
+}
+
+const MAX_POSTS = 1000;
+const MAX_REPLIES = 1000;
+
+export const calculateParticipation = (
```

Comment:
It seems like we could add in here a check against the current date and time vs `postDeadline` and `replyDeadline` which could factor into setting `canPost` and `canReply`.  That will at least give us an initial (but a bit weak) form of scheduling for the directed discussions.  We'd just have to also ensure that the PostDeadline and ReplyDeadlines are displayed to the student when these are set.